src/autoschedulers/anderson2021/LoopNest.h

```diff
@@ -0,0 +1,601 @@
+/** This file defines the LoopNest, which is our
+ * representation of a Halide schedule, and contains methods to
+ * generate candidates for scheduling as well as extract a
+ * featurization that can be used to cost each candidate. */
+
+#ifndef LOOP_NEST_H
+#define LOOP_NEST_H
+
+#include "ASLog.h"
+#include "CostModel.h"
+#include "FunctionDAG.h"
+#include "GPULoopInfo.h"
+#include "GPUMemInfo.h"
+#include "PerfectHashMap.h"
+#include "SearchSpaceOptions.h"
+#include "Statistics.h"
+#include "ThreadInfo.h"
+#include "Tiling.h"
+#include <set>
+#include <vector>
+
+namespace Halide {
+namespace Internal {
+namespace Autoscheduler {
+
+template<typename T>
+using NodeMap = PerfectHashMap<FunctionDAG::Node, T>;
+
+template<typename T>
+using StageMap = PerfectHashMap<FunctionDAG::Node::Stage, T>;
+
+enum GPU_parallelism { block,
```

Comment:
Style nit: enums in Halide generally begin with Uppercase. 

Also, using `enum class` tends to be preferred for most new enums at this point (though this definitely isn't required, sometimes terseness or other usage factors makes `enum class` undesirable.